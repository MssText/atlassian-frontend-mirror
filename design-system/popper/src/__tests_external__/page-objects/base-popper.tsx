import {
  CypressType,
  InProductTestPageObject,
} from '@atlaskit/in-product-testing';

export class BasePopperPageObject extends InProductTestPageObject {
  constructor(cy: CypressType) {
    super(cy);
  }

  public assertPopperStyle = (
    elementSelector: keyof HTMLElementTagNameMap | string,
  ) => {
    return this.cy.get(elementSelector).should('be.visbile');
  };
}
