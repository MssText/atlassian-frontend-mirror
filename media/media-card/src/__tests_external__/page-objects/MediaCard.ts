// complains about devDependencies instead of dependencies
// eslint-disable-next-line import/no-extraneous-dependencies
import { cy } from 'local-cypress';

export class MediaCardPageObject {
  constructor(private cy: Cypress.cy) {}

  public expectCardReady(numOfCards: number) {
    return this.cy
      .get('[data-test-media-name]')
      .should('have.length', numOfCards)
      .each((element) => {
        cy.wrap(element)
          // NOTE: `be.visible` is used to ensure the file is still visible
          // whilst it is being uploaded.
          .should('be.visible')
          .should('have.attr', 'data-test-status', 'complete');
      });
  }

  public findCardNth(nth: number) {
    return this.cy
      .get('[data-test-media-name][data-test-status="complete"]')
      .should('have.length.at.least', nth)
      .should('be.visible')
      .eq(nth - 1);
  }
}
