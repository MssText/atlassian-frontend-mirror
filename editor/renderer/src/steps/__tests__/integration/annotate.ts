import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import {
  mountRenderer,
  goToRendererTestingExample,
} from '../../../__tests__/__helpers/testing-example-helpers';
import { selectors } from '../../../__tests__/__helpers/page-objects/_renderer';
import { annotate } from '../../../__tests__/__helpers/page-objects/_actions';

const CHAR_WIDTH = 6;

BrowserTestCase(
  `Can't create an annotation on a text selection that contains inline nodes`,
  { skip: ['ie', 'edge', 'firefox'] },
  async (client: any, testName: string) => {
    const page = await goToRendererTestingExample(client);
    await mountRenderer(
      page,
      { withRendererActions: true },
      {
        version: 1,
        type: 'doc',
        content: [
          {
            type: 'paragraph',
            content: [
              {
                type: 'text',
                text: 'hello ',
              },
              {
                type: 'emoji',
                attrs: {
                  shortName: ':cactus:',
                  id: '1f335',
                  text: 'ðŸŒµ',
                },
              },
              {
                type: 'text',
                text: ' mate',
              },
            ],
          },
        ],
      },
    );

    await page.simulateUserSelection(
      `${selectors.document} > p`,
      `${selectors.document} > p`,
      element => (element === 'start' ? CHAR_WIDTH : 0),
    );

    const result = await annotate(page, '1234');
    expect(result).toBeUndefined;
  },
);

BrowserTestCase(
  'Can create an annotation on a basic text selection',
  { skip: ['ie', 'edge', 'firefox'] },
  async (client: any, testName: string) => {
    const page = await goToRendererTestingExample(client);
    await mountRenderer(
      page,
      { withRendererActions: true },
      {
        type: 'doc',
        version: 1,
        content: [
          {
            type: 'paragraph',
            content: [
              {
                type: 'text',
                text: 'Hello, World',
              },
            ],
          },
        ],
      },
    );

    await page.simulateUserSelection(
      `${selectors.document} > p`,
      `${selectors.document} > p`,
      element => (element === 'start' ? CHAR_WIDTH : 0),
    );

    const result = await annotate(page, '1234');
    expect(result).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  'Can create an annotation on a text selection over two paragraphs',
  { skip: ['ie', 'edge', 'firefox'] },
  async (client: any, testName: string) => {
    const page = await goToRendererTestingExample(client);
    await mountRenderer(
      page,
      { withRendererActions: true },
      {
        type: 'doc',
        version: 1,
        content: [
          {
            type: 'paragraph',
            content: [
              {
                type: 'text',
                text:
                  'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean mi nisl, venenatis eget auctor vitae, venenatis quis lorem. Suspendisse maximus tortor vel dui tincidunt cursus. Vestibulum magna nibh, auctor non auctor id, finibus vitae orci. Nulla viverra ipsum et nunc fringilla ultricies. Pellentesque vitae felis molestie justo finibus accumsan. Suspendisse potenti. Nulla facilisi. Integer dignissim quis velit quis elementum. Sed sit amet varius ante. Duis vestibulum porta augue eu laoreet. Morbi id risus et augue sollicitudin aliquam. In et ligula dolor. Nam ac aliquet diam.',
              },
            ],
          },
          {
            type: 'paragraph',
            content: [
              {
                type: 'text',
                text:
                  'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean mi nisl, venenatis eget auctor vitae, venenatis quis lorem. Suspendisse maximus tortor vel dui tincidunt cursus. Vestibulum magna nibh, auctor non auctor id, finibus vitae orci. Nulla viverra ipsum et nunc fringilla ultricies. Pellentesque vitae felis molestie justo finibus accumsan. Suspendisse potenti. Nulla facilisi. Integer dignissim quis velit quis elementum. Sed sit amet varius ante. Duis vestibulum porta augue eu laoreet. Morbi id risus et augue sollicitudin aliquam. In et ligula dolor. Nam ac aliquet diam.',
              },
            ],
          },
        ],
      },
    );

    await page.simulateUserSelection(
      `${selectors.document} > p`,
      `${selectors.document} > p:nth-child(2)`,
      element => (element === 'start' ? CHAR_WIDTH : -(CHAR_WIDTH * 5)),
    );

    const result = await annotate(page, '1234');
    expect(result).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  'Can create an annotation on a text selection over a decision item',
  { skip: ['ie', 'edge', 'firefox'] },
  async (client: any, testName: string) => {
    const page = await goToRendererTestingExample(client);
    await mountRenderer(
      page,
      { withRendererActions: true },
      {
        type: 'doc',
        version: 1,
        content: [
          {
            type: 'paragraph',
            content: [
              {
                type: 'text',
                text: 'text before',
              },
            ],
          },
          {
            type: 'decisionList',
            attrs: {
              localId: 'local-uuid',
            },
            content: [
              {
                type: 'decisionItem',
                attrs: {
                  localId: 'local-uuid',
                  state: 'DECIDED',
                },
                content: [
                  {
                    type: 'text',
                    text: 'hello, mate',
                  },
                ],
              },
            ],
          },
          {
            type: 'paragraph',
            content: [
              {
                type: 'text',
                text: 'text after',
              },
            ],
          },
        ],
      },
    );

    await page.simulateUserSelection(
      `${selectors.document} > p`,
      `${selectors.document} > p:last-child`,
      element => (element === 'start' ? CHAR_WIDTH : 0),
    );

    const result = await annotate(page, '1234');
    expect(result).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  'Can create an annotation on a text selection over a task item',
  { skip: ['ie', 'edge', 'firefox'] },
  async (client: any, testName: string) => {
    const page = await goToRendererTestingExample(client);
    await mountRenderer(
      page,
      { withRendererActions: true },
      {
        type: 'doc',
        version: 1,
        content: [
          {
            type: 'paragraph',
            content: [
              {
                type: 'text',
                text: 'text before',
              },
            ],
          },
          {
            type: 'taskList',
            attrs: {
              localId: 'local-uuid',
            },
            content: [
              {
                type: 'taskItem',
                attrs: {
                  localId: 'local-uuid',
                  state: 'TODO',
                },
                content: [
                  {
                    type: 'text',
                    text: 'hello, mate',
                  },
                ],
              },
            ],
          },
          {
            type: 'paragraph',
            content: [
              {
                type: 'text',
                text: 'text after',
              },
            ],
          },
        ],
      },
    );

    await page.simulateUserSelection(
      `${selectors.document} > p`,
      `${selectors.document} > p:last-child`,
      element => (element === 'start' ? CHAR_WIDTH : 0),
    );

    const result = await annotate(page, '1234');
    expect(result).toMatchCustomDocSnapshot(testName);
  },
);
