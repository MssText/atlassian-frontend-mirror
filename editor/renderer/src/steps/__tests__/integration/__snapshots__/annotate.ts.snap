// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can create an annotation on a basic text selection 1`] = `
Object {
  "doc": Object {
    "content": Array [
      Object {
        "content": Array [
          Object {
            "text": "H",
            "type": "text",
          },
          Object {
            "marks": Array [
              Object {
                "attrs": Object {
                  "annotationType": "inlineComment",
                  "id": "1234",
                },
                "type": "annotation",
              },
            ],
            "text": "ello, World",
            "type": "text",
          },
        ],
        "type": "paragraph",
      },
    ],
    "type": "doc",
    "version": 1,
  },
  "step": Object {
    "from": 2,
    "mark": Object {
      "attrs": Object {
        "annotationType": "inlineComment",
        "id": "1234",
      },
      "type": "annotation",
    },
    "stepType": "addMark",
    "to": 13,
  },
}
`;

exports[`Can create an annotation on a text selection over a decision item 1`] = `
Object {
  "doc": Object {
    "content": Array [
      Object {
        "content": Array [
          Object {
            "text": "t",
            "type": "text",
          },
          Object {
            "marks": Array [
              Object {
                "attrs": Object {
                  "annotationType": "inlineComment",
                  "id": "1234",
                },
                "type": "annotation",
              },
            ],
            "text": "ext before",
            "type": "text",
          },
        ],
        "type": "paragraph",
      },
      Object {
        "attrs": Object {
          "localId": "local-uuid",
        },
        "content": Array [
          Object {
            "attrs": Object {
              "localId": "local-uuid",
              "state": "DECIDED",
            },
            "content": Array [
              Object {
                "marks": Array [
                  Object {
                    "attrs": Object {
                      "annotationType": "inlineComment",
                      "id": "1234",
                    },
                    "type": "annotation",
                  },
                ],
                "text": "hello, mate",
                "type": "text",
              },
            ],
            "type": "decisionItem",
          },
        ],
        "type": "decisionList",
      },
      Object {
        "content": Array [
          Object {
            "marks": Array [
              Object {
                "attrs": Object {
                  "annotationType": "inlineComment",
                  "id": "1234",
                },
                "type": "annotation",
              },
            ],
            "text": "text after",
            "type": "text",
          },
        ],
        "type": "paragraph",
      },
    ],
    "type": "doc",
    "version": 1,
  },
  "step": Object {
    "from": 2,
    "mark": Object {
      "attrs": Object {
        "annotationType": "inlineComment",
        "id": "1234",
      },
      "type": "annotation",
    },
    "stepType": "addMark",
    "to": 39,
  },
}
`;

exports[`Can create an annotation on a text selection over a task item 1`] = `
Object {
  "doc": Object {
    "content": Array [
      Object {
        "content": Array [
          Object {
            "text": "t",
            "type": "text",
          },
          Object {
            "marks": Array [
              Object {
                "attrs": Object {
                  "annotationType": "inlineComment",
                  "id": "1234",
                },
                "type": "annotation",
              },
            ],
            "text": "ext before",
            "type": "text",
          },
        ],
        "type": "paragraph",
      },
      Object {
        "attrs": Object {
          "localId": "local-uuid",
        },
        "content": Array [
          Object {
            "attrs": Object {
              "localId": "local-uuid",
              "state": "TODO",
            },
            "content": Array [
              Object {
                "marks": Array [
                  Object {
                    "attrs": Object {
                      "annotationType": "inlineComment",
                      "id": "1234",
                    },
                    "type": "annotation",
                  },
                ],
                "text": "hello, mate",
                "type": "text",
              },
            ],
            "type": "taskItem",
          },
        ],
        "type": "taskList",
      },
      Object {
        "content": Array [
          Object {
            "marks": Array [
              Object {
                "attrs": Object {
                  "annotationType": "inlineComment",
                  "id": "1234",
                },
                "type": "annotation",
              },
            ],
            "text": "text after",
            "type": "text",
          },
        ],
        "type": "paragraph",
      },
    ],
    "type": "doc",
    "version": 1,
  },
  "step": Object {
    "from": 2,
    "mark": Object {
      "attrs": Object {
        "annotationType": "inlineComment",
        "id": "1234",
      },
      "type": "annotation",
    },
    "stepType": "addMark",
    "to": 39,
  },
}
`;

exports[`Can create an annotation on a text selection over two paragraphs 1`] = `
Object {
  "doc": Object {
    "content": Array [
      Object {
        "content": Array [
          Object {
            "text": "L",
            "type": "text",
          },
          Object {
            "marks": Array [
              Object {
                "attrs": Object {
                  "annotationType": "inlineComment",
                  "id": "1234",
                },
                "type": "annotation",
              },
            ],
            "text": "orem ipsum dolor sit amet, consectetur adipiscing elit. Aenean mi nisl, venenatis eget auctor vitae, venenatis quis lorem. Suspendisse maximus tortor vel dui tincidunt cursus. Vestibulum magna nibh, auctor non auctor id, finibus vitae orci. Nulla viverra ipsum et nunc fringilla ultricies. Pellentesque vitae felis molestie justo finibus accumsan. Suspendisse potenti. Nulla facilisi. Integer dignissim quis velit quis elementum. Sed sit amet varius ante. Duis vestibulum porta augue eu laoreet. Morbi id risus et augue sollicitudin aliquam. In et ligula dolor. Nam ac aliquet diam.",
            "type": "text",
          },
        ],
        "type": "paragraph",
      },
      Object {
        "content": Array [
          Object {
            "marks": Array [
              Object {
                "attrs": Object {
                  "annotationType": "inlineComment",
                  "id": "1234",
                },
                "type": "annotation",
              },
            ],
            "text": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean mi nisl, venenatis eget auctor vitae, venenatis quis lorem. Suspendisse maximus tortor vel dui tincidunt cursus. Vestibulum magna nibh, auctor non auctor id, finibus vitae orci. Nulla viverra ipsum et nunc fringilla ultricies. Pellentesque vitae felis molestie justo finibus accumsan. Suspendisse potenti. Nulla facilisi. Integer dignissim quis velit quis elementum. Sed sit amet varius ante. Duis vestibulum porta augue eu laoreet. Morbi id risus et augue sollicitudin aliquam. In et ligula dolor. Nam ac aliquet diam.",
            "type": "text",
          },
        ],
        "type": "paragraph",
      },
    ],
    "type": "doc",
    "version": 1,
  },
  "step": Object {
    "from": 2,
    "mark": Object {
      "attrs": Object {
        "annotationType": "inlineComment",
        "id": "1234",
      },
      "type": "annotation",
    },
    "stepType": "addMark",
    "to": 1169,
  },
}
`;
